src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx

```diff
@@ -101,9 +102,69 @@ const RuleEditor = createReactClass({
     Promise.all(promises).then(([environments, rule]) => {
       this.setState({environments, rule});
     });
-  },
+  }
+
+  pollHandler = async quitTime => {
+    if (Date.now() > quitTime) {
+      addErrorMessage(t('Looking for that channel took too long :('));
+      this.setState({loading: false});
+    } else {
+      const {api, organization, project} = this.props;
+      const {uuid} = this.state;
+      const origRule = {...this.state.rule};
+
+      const response = await api
+        .requestPromise(
+          `/projects/${organization.slug}/${project.slug}/rule-task/${uuid}/`
+        )
+        .catch(() => {
+          addErrorMessage(t('An error occurred'));
+          this.setState({loading: false});
+        });
+
+      const {status, rule, error} = response;
+
+      if (status === 'pending') {
+        setTimeout(() => {
+          this.pollHandler(quitTime);
+        }, 1000);
+        return;
+      }
+
+      if (status === 'failed') {
+        this.setState({
+          error: {actions: [error]} || {__all__: 'Unknown error'},
```

Comment:
@MeredithAnya Why do we set the state of the error here but not in the other error handler above?

Reply:
This is to replicate the similar behavior we have now: 
<img width="819" alt="Screen Shot 2020-02-05 at 10 08 06 AM" src="https://user-images.githubusercontent.com/15368179/73869592-6da53e80-47ff-11ea-9213-d28b831d2240.png">
it's kind of hacky since I'm hardcoding in the `actions` part but we don't have the validation error being passed through like we do when it doesn't have to run the task

Reply:
@MeredithAnya Got it. BTW, I don't think the default condition of `{__all__: 'Unknown error'}` is ever going to get used because `{actions: [error]}` is truthy.